include/alpaka/test/Check.hpp

```diff
@@ -14,7 +14,16 @@
         {                                                                                                             \
             if(!(expression))                                                                                         \
             {                                                                                                         \
-                acc.cout << "ALPAKA_CHECK failed because '!(" << #expression << ")'\n";                               \
+                /*if constexpr(alpaka::accMatchesTags<   TODO                                                         \
+                                 decltype(acc),                                                                       \
+                                 alpaka::TagCpuSyclIntel,                                                             \
+                                 alpaka::TagFpgaSyclIntel,                                                            \
+                                 alpaka::TagFpgaSyclXilinx,                                                           \
+                                 alpaka::TagGpuSyclIntel,                                                             \
+                                 alpaka::TagGenericSycl>)                                                             \
+                    acc.cout << "ALPAKA_CHECK failed because '!(" << #expression << ")'\n";                           \
+                else                                                                                                  \
+                    printf("ALPAKA_CHECK failed because '!(%s)'\n", #expression);*/                                   \
                 success = false;                                                                                      \
             }                                                                                                         \
```

Comment:
You will need to fix `ALPAKA_CHECK` here like i sketched it in the commented code.

Reply:
I don't understand the reason of this change. Keeping it as it was, for SYCL it will call the `printf` we implemented in [`core`](https://github.com/Parsifal-2045/alpaka/blob/8220990a43429fd5a9ecd4479ea120e607095d46/include/alpaka/core/Sycl.hpp#L37) if this is the problem

Reply:
Right. But the `printf` you implemented uses a OneAPI extension and probably does not work on other SYCL implementations. @j-stephan implemented device side printing using this `cout` member on the accelerator object for this reason.  But I will let him comment on this issue.

Reply:
I'm fine with using Intel's `printf`. Looking at the current state of the SYCL ecosystem it is more or less becoming Intel-exclusive anyway. We *might* care about [neoSYCL](https://github.com/Tohoku-University-Takizawa-Lab/neoSYCL) in the future but we'll cross that bridge when we reach it.